pkg/eks/api/api.go

```diff
@@ -89,4 +89,5 @@ type ClusterAddons struct {
 // AddonIAM provides an addon for the AWS IAM integration
 type AddonIAM struct {
 	PolicyAmazonEC2ContainerRegistryPowerUser bool
+	AutoScalingGroup                          bool
```

Comment:
perhaps the `PolicyAmazonEC2` is a bit over the top, but this would make it consistent: 
```suggestion
	PolicyAutoScaling                          bool
```